TFUtil.py

```diff
@@ -6667,3 +6667,32 @@ def get_positional_encoding(num_channels, length=None, position=None, min_timesc
   signal = tf.concat([tf.sin(scaled_time), tf.cos(scaled_time)], axis=1)
   signal = tf.pad(signal, [[0, 0], [0, num_channels % 2]])  # (length, channels)
   return signal
+
+def check_contains_only_det_ops(print_trace=False):
+  """
+  :param bool print_trace: prints a trace of which ops are checked
+  :return:  are all ops used on the available devices deterministic?
+  """
+  # List of non deterministic (NDT) ops (depending on version and device)
+  non_deterministic_ops = {
+    "Mean": ["1.5.0", ["GPU"]]
```

Comment:
In general: Do not use a list, but a tuple for things like this.
But in this case (and also almost in general, in most cases), I think you even should introduce some helper class, which makes it more descriptive what the individual fields mean (e.g. min_version, dev_type, or so).
But then, maybe also this makes the whole code too complicated. Maybe just do not introduce such dict at all here, and directly do the check in the `for op in op_set` loop below. I.e. just do a big `if op == "Mean": ... elif ...` etc.